lapce-proxy/src/terminal.rs

```diff
@@ -41,62 +43,20 @@ pub struct Terminal {
 impl Terminal {
     pub fn new(
         term_id: TermId,
-        cwd: Option<PathBuf>,
-        env: Option<HashMap<String, String>>,
-        shell: String,
+        profile: TerminalProfile,
         width: usize,
         height: usize,
     ) -> Terminal {
         let poll = mio::Poll::new().unwrap();
         let mut config = TermConfig::default();
-        config.pty_config.working_directory =
-            if cwd.is_some() && cwd.clone().unwrap().exists() {
-                cwd
-            } else {
-                lapce_core::directory::Directory::home_dir()
-            };
-        config.env = env.unwrap_or_default();
-
-        let shell = shell.trim();
-        let flatpak_use_host_terminal = flatpak_should_use_host_terminal();
-
-        if !shell.is_empty() || flatpak_use_host_terminal {
-            let mut parts = shell.split(' ');
-
-            if flatpak_use_host_terminal {
-                let flatpak_spawn_path = "/usr/bin/flatpak-spawn".to_string();
-                let host_shell = flatpak_get_default_host_shell();
-
-                let args = if shell.is_empty() {
-                    vec![
-                        "--host".to_string(),
-                        "--env=TERM=alacritty".to_string(),
-                        host_shell,
-                    ]
-                } else {
-                    vec![
-                        "--host".to_string(),
-                        "--env=TERM=alacritty".to_string(),
-                        host_shell,
-                        "-c".to_string(),
-                        shell.to_string(),
-                    ]
-                };
-
-                config.pty_config.shell = Some(Program::WithArgs {
-                    program: flatpak_spawn_path,
-                    args,
-                })
-            } else {
-                let program = parts.next().unwrap();
-                if let Ok(p) = which::which(program) {
-                    config.pty_config.shell = Some(Program::WithArgs {
-                        program: p.to_str().unwrap().to_string(),
-                        args: parts.map(|p| p.to_string()).collect::<Vec<String>>(),
-                    })
-                }
-            }
```

Comment:
Why do you remove the flatpack related stuff?

Reply:
Because it was a hack, with terminal profiles user can configure if they want to use host tools or not